downloader/src/main/kotlin/VersionControlSystem.kt

```diff
@@ -238,24 +238,32 @@ abstract class VersionControlSystem {
             }
         }
 
-        try {
-            workingTree.guessRevisionName(pkg.id.name, pkg.id.version).also {
-                if (revisionCandidates.add(it)) {
-                    log.info {
-                        "Adding $type revision '$it' (guessed from package '${pkg.id.name}' and version " +
-                                "'${pkg.id.version}') as a candidate."
+        fun addGuessedRevision(project: String, version: String): Boolean {
+            try {
+                workingTree.guessRevisionName(project, version).also {
+                    if (revisionCandidates.add(it)) {
+                        log.info {
+                            "Adding $type revision '$it' (guessed from package '$project' and version " +
+                                    "'$version') as a candidate."
+                        }
                     }
                 }
-            }
-        } catch (e: IOException) {
-            e.showStackTrace()
 
-            log.info {
-                "No $type revision for package '${pkg.id.name}' and version '${pkg.id.version}' found: " +
-                    e.collectMessagesAsString()
+                return true
+            } catch (e: IOException) {
+                e.showStackTrace()
+
+                log.info {
+                    "No $type revision for package '$project' and version '$version' found: " +
+                            e.collectMessagesAsString()
+                }
+
+                return false
```

Comment:
Nit: Could lift out the `return` to before `try`, and then also make this a function expression.